k8s-service-account-config-to-ssm/kube_config_generator.py

```diff
@@ -0,0 +1,276 @@
+import base64
+import boto3
+import getopt
+import sys
+import re
+import subprocess
+import logging
+from kubernetes import client, config
+from kubernetes.client.rest import ApiException
+
+# define constants
+AWS_ROLE_CLOUD_ADMIN = "arn:aws:iam::738063116313:role/CloudAdmin"
+AWS_ROLE_ADFS_ADMIN = "arn:aws:iam::454234050858:role/ADFS-Admin"
+AWS_PARAMETER_NAME = "/managed/deploy/kube-config"
+AWS_PROFILE = "saml"
+AWS_REGION = "eu-central-1"
+CAPABILITY_AWS_ROLE_SESSION = "kube-config-paramstore"
+KUBERNETES_CONTEXT = "hellman-saml"
+SERVICE_ACCOUNT_NAMESPACE = "kube-system"
+
+
+def assume_saml_role(role_to_assume: str) -> bool:
+    """
+    Use the saml2aws utility to assume a specific SAML role in the AWS environment
+    """
+    logging.info(f"Assuming the SAML Role {role_to_assume}.")
+    exec_command: str = f"saml2aws login --role={role_to_assume} --force --skip-prompt"
+    proc_result: subprocess.CompletedProcess = subprocess.run(
+        exec_command.split(" "), stdout=subprocess.DEVNULL
+    )
+    if proc_result.returncode == 0:
+        logging.info("Role assumed succesfully.")
+    else:
+        logging.warn("Role failed to be assumed.")
+    return proc_result.returncode == 0
+
+
+def create_k8s_kube_config(service_account_name: str, namespace: str) -> str:
+    """
+    Create a string hold the kubeconfig definition.
+    """
+    kube_api = client.CoreV1Api()
+    list_secrets = kube_api.list_namespaced_secret(SERVICE_ACCOUNT_NAMESPACE)
+    for secret in list_secrets.items:
+        if re.search(f"^{service_account_name}", secret.metadata.name):
+            kube_token_b64: str = secret.data["token"]
+            break
+
+    if kube_token_b64 != "":
```

Comment:
If "if re.search(f"^{service_account_name}", secret.metadata.name)" is False, then kube_token_b64 is never created, in whihc case it will have the value None. 

Hence, I think it would be better to rewrite this part to:

if kube_token_b64 is not None or kube_token_b64 != "":

Reply:
I've left the code logic as is but have added a line at the beginning of the function that initialises the variable as an empty string.